---
layout: page
title: Links
permalink: /links/
---

* TOC
{:toc}

## Bioinformatics

### Blogs

- [陈国强](http://blog.shsmu-path.com.cn/)
- [薛宇](http://blog.sciencenet.cn/home.php?mod=space&uid=404304)
- [饶毅](http://blog.sciencenet.cn/?2237)
- [雁起平沙](http://yanping.me/cn/)
- [萧之逸](http://lwzhanghz.blog.163.com/blog/#m=0)
- [云之南](http://fhqdddddd.blog.163.com/)
- [陈连福](http://www.chenlianfu.com/)
- [鲁白](http://blog.sciencenet.cn/home.php?mod=space&uid=393255)
- [YGC](http://guangchuangyu.github.io/)
- [yixf](http://yixf.name/)
- [azalea](http://azaleasays.com/)
- [David Langenberger](http://hoffmann.bioinf.uni-leipzig.de/LIFE/David.html)
- [Jiang Li](https://sites.google.com/site/riverlee2008/)
- Heng Li: 1) [sourceforge](http://lh3lh3.users.sourceforge.net/) 2) [github](http://lh3.github.io/)
- [onetipperday](http://onetipperday.blogspot.com/)
- [Bergman Lab](http://bergmanlab.ls.manchester.ac.uk/?page_id=45)
- [Ewan's blog](http://genomeinformatician.blogspot.com/)
- [Xiaole Shirley Liu's Blog Site](http://www.longwoodgenomics.org/)
- [Yokofakun](http://plindenbaum.blogspot.com/)
- [Puriney](http://joseph.yy.blog.163.com/)
- [RNA-seq Bolg](http://www.rna-seqblog.com/)
- [Simply Statistics](http://simplystatistics.org/)
- [What You're Doing Is Rahter Desperate](https://nsaunders.wordpress.com/)
- [Win-Vector Blog](http://www.win-vector.com/blog/)
- [Genomes Unzipped](http://genomesunzipped.org/)
- [MassGenomics](http://massgenomics.org/)
- [CoreGenomics](http://core-genomics.blogspot.com/)


### Softwares

**GWAS**

- [PLINK (v1.07)](http://zzz.bwh.harvard.edu/plink/): whole genome association analysis toolset
- [PLINK2 (v1.9 beta)](https://www.cog-genomics.org/plink2): a comprehensive update to Shaun Purcell's PLINK
- [GCTA](http://cnsgenomics.com/software/gcta/): to estimate the proportion of phenotypic variance explained by genome- or chromosome-wide SNPs for complex traits
- [SMR](http://cnsgenomics.com/software/smr/index.html): association between gene expression and a complex trait
- [GSMR](http://cnsgenomics.com/software/gsmr/gsmr.html): causal association between a risk factor and disease
- [MACH](http://csg.sph.umich.edu/abecasis/mach/index.html): resolve long haplotypes or infer missing genotypes
- [IMPUTE2](http://mathgen.stats.ox.ac.uk/impute/impute_v2.html): genotype imputation and haplotype phasing program
- [METAL](http://genome.sph.umich.edu/wiki/METAL_Program): meta-analysis of genomewide association scans
- [DEPICT](https://data.broadinstitute.org/mpg/depict/): an integrative tool that based on predicted gene functions systematically prioritizes the most likely causal genes at associated loci, highlights enriched pathways, and identifies tissues/cell types where genes from associated loci are highly expressed
- [Beagle](https://faculty.washington.edu/browning/beagle/beagle.html): genotype calling, genotype phasing, imputation of ungenotyped markers, and identity-by-descent segment detection
- [haploview](https://www.broadinstitute.org/haploview/haploview): analysis and visualization of LD and haplotype maps
- [HaploReg](http://archive.broadinstitute.org/mammals/haploreg/haploreg.php): annotations of the noncoding genome at variants
- [BOLT-LMM](https://data.broadinstitute.org/alkesgroup/BOLT-LMM/): BOLT-LMM algorithm for mixed model association testing, and the BOLT-REML algorithm for variance components analysis 

**ChIP-seq**

- [bwa](http://bio-bwa.sourceforge.net/)
- [bowtie](http://bowtie-bio.sourceforge.net/index.shtml)
- [MACS](http://liulab.dfci.harvard.edu/MACS/)
- [MACS2](https://github.com/taoliu/MACS/)
- [SPP](http://compbio.med.harvard.edu/Supplements/ChIP-seq/)
- [HPeak](http://www.sph.umich.edu/csg/qin/HPeak/Readme.html)
- [meme-chip](http://meme.nbcr.net/meme/doc/meme-chip.html)
- [GimmeMotif](http://131.174.198.125/bioinfo/gimmemotifs/)
- [homer](http://biowhat.ucsd.edu/homer/)
- [CEAS](http://liulab.dfci.harvard.edu/CEAS/)
- [STAMP](http://www.benoslab.pitt.edu/stamp/)

**RNA-seq**

- [tophat](http://tophat.cbcb.umd.edu/)
- [STAR](https://github.com/alexdobin/STAR)
- [HISAT](https://ccb.jhu.edu/software/hisat/index.shtml)
- [StringTie](http://www.ccb.jhu.edu/software/stringtie/)
- [RSeQC](http://rseqc.sourceforge.net/)
- [RNA-SeQC](http://www.broadinstitute.org/cancer/cga/rna-seqc)
- [Cufflinks](http://cufflinks.cbcb.umd.edu/)
- [Scripture](https://www.broadinstitute.org/software/scripture/)
- [HTSeq](http://www-huber.embl.de/users/anders/HTSeq/doc/overview.html)
- [featurecount](http://bioinf.wehi.edu.au/featureCounts/)
- [GFold](http://web.tongji.edu.cn/~zhanglab/GFOLD/)
- [DESeq](http://bioconductor.org/packages/2.12/bioc/html/DESeq.html)
- [DESeq2](http://www.bioconductor.org/packages/release/bioc/html/DESeq2.html)
- [DEGseq](http://bioinfo.au.tsinghua.edu.cn/software/degseq/)
- [DEXSeq](http://www.bioconductor.org/packages/release/bioc/html/DEXSeq.html)
- [eXpress](http://bio.math.berkeley.edu/eXpress/overview.html)
- [RSEM](http://deweylab.biostat.wisc.edu/rsem/)
- [MISO]( http://genes.mit.edu/burgelab/miso/)
- [MATS](http://rnaseq-mats.sourceforge.net/)
- [MMSEQ](http://bgx.org.uk/software/mmseq.html)
- [CIRCexplorer](https://github.com/YangLab/CIRCexplorer)
- [CIRCfinder](https://github.com/YangLab/CIRCfinder)

**WGS/WXS**

- [GATK](https://www.broadinstitute.org/gatk/)
- [MuTect](http://www.broadinstitute.org/cancer/cga/MuTect)
- [Pindel](http://gmt.genome.wustl.edu/packages/pindel/)
- [CREST](http://www.ncbi.nlm.nih.gov/pubmed/21666668)
- [SomaticSniper](http://gmt.genome.wustl.edu/packages/somatic-sniper/index.html)
- [VarScan2](http://varscan.sourceforge.net/)
- [Strelka](https://sites.google.com/site/strelkasomaticvariantcaller/)
- [JointSNVMix](https://code.google.com/p/joint-snv-mix/)
- [MutSig](https://www.broadinstitute.org/cancer/cga/mutsig)
- [InVEx](http://www.broadinstitute.org/cancer/cga/invex/)
- [Oncotator](http://www.broadinstitute.org/oncotator/)
- [bradkDancer](http://breakdancer.sourceforge.net/)
- [SciClone](https://github.com/genome/sciclone)


**SNP array**

- OncoSNP:

**microarray**

- RMA:
- MAS5.0:
- GCRMA:

**Common Tools**

- [samtools](http://samtools.sourceforge.net/)
- [samstat](http://samstat.sourceforge.net/)
- [bedtools](http://www.broadinstitute.org/cancer/cga/rna-seqc)
- [bedops](https://code.google.com/p/bedops/)
- [Fastqc](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/)
- [Fastx_toolkit](http://hannonlab.cshl.edu/fastx_toolkit/)
- [trimmomatic](http://www.usadellab.org/cms/index.php?page=trimmomatic)
- [Picard](http://broadinstitute.github.io/picard/)
- [ucsc tools](http://hgdownload.cse.ucsc.edu/admin/exe/)
- [CIRCOS](http://mkweb.bcgsc.ca/circos)
- [SRAtoolkit](https://github.com/ncbi/sratoolkit)
- [Entrez Direct](http://www.ncbi.nlm.nih.gov/news/02-06-2014-entrez-direct-released/)
- [GenePattern](http://www.broadinstitute.org/cancer/software/genepattern/)

### Bioinformatic forum

- [Biostars](http://www.biostars.org/)
- [Seqanswers](http://seqanswers.com/)
- [links directory](http://bioinformatics.ca/links_directory/)

### Personal website

**Hematology**

- [Daniel G Tenen](http://hsci.harvard.edu/people/daniel-g-tenen)
- [Ellen Rothenberg](http://www.bbe.caltech.edu/content/ellen-rothenberg)
- [James C. Mulloy](http://www.cincinnatichildrens.org/bio/m/james-mulloy/)
- [Constanze Bonifer](http://www.birmingham.ac.uk/staff/profiles/cancer/bonifer-constanze.aspx)
- [Bertie G?ttgens](http://www.stemcells.cam.ac.uk/researchers/principal-investigators/bertie-gottgens)
- [DongEr Zhang](http://labs.biology.ucsd.edu/zhang/)
- [Jinghui Zhang](http://www.stjude.org/stjude/v/index.jsp?vgnextoid=39254a82840f7210VgnVCM1000001e0215acRCRD)
- [Margaret A. Goodell](https://www.bcm.edu/research/labs/goodell/)
- [Robert G Roeder](http://www.rockefeller.edu/research/faculty/labheads/RobertRoeder/)
- [Stephen D. Nimer](http://uhealthsystem.com/doctors/profile/134105)

**Genome Research**

- [Eric S. Lander](http://www.broadinstitute.org/history-leadership/scientific-leadership/core-members/eric-s-lander)
- [Alexander Stark](http://www.starklab.org/)
- [Yi Zhang](http://zhanglab.tch.harvard.edu/lab.htm)
- [Howard Chang](http://changlab.stanford.edu/)
- [John Stamatoyannopoulos](http://www.stamlab.org/)
- [Bin Ren](http://bioinformatics-renlab.ucsd.edu/rentrac/)
- [Zhiping Weng](http://zlab.umassmed.edu/zlab/index.shtml)
- [Stunnenberg](http://www.ncbi.nlm.nih.gov/pubmed?cmd=search&term=Stunnenberg+HG%5Bau%5D&dispmax=50)
- [Xiaole(Shirley) Liu](http://liulab.dfci.harvard.edu/)
- [Young Lab](http://younglab.wi.mit.edu/)
- [Wang Ting](http://wang.wustl.edu/)

**Computational Biologist**

- [Yang Li](http://www.picb.ac.cn/rnomics/yanglab/)
- [Istvan Albert ](http://www.personal.psu.edu/iua1/)
- [Xuegong Zhang]( http://bioinfo.au.tsinghua.edu.cn/member/xzhang/XZhang_Chinese.html)
- [Ao Ping](http://systemsbiology.sjtu.edu.cn/)
- [rafalab]( http://rafalab.dfci.harvard.edu/)
- [Lior Pachter](https://math.berkeley.edu/~lpachter/index.html)
- [Cole Trapnell](http://cole-trapnell-lab.github.io/)
- [Steven Salzberg](http://salzberg-lab.org/)
- [Yarden Katz](http://www.mit.edu/~yarden/)
- [Wolfgang Huber](http://www.embl.de/research/units/genome_biology/huber/index.html)
- [Mike Snyder](http://snyderlab.stanford.edu/)
- [Wong Yong Wang](http://web.stanford.edu/group/wonglab/)
- [Peter Visshcer](http://cnsgenomics.com/index.html)

**My lab**

- [王侃侃](http://daoshi.shsmu.edu.cn/Pages/TeacherInformationView.aspx?uid=A32E6543-1E2C-4136-91D8-C5E48D6396BC&from=s&pId=&tId=720)
- [方海](http://www.cs.bris.ac.uk/~hfang/)


### Journal
- [Cell](http://www.cell.com/)
- [Science](http://www.sciencemag.org/)
- [Nature](http://www.nature.com/)
- [Blood](http://bloodjournal.hematologylibrary.org/)
- [Leukemia](http://www.nature.com/leu/index.html)
- [PNAS](http://www.pnas.org/)
- [NEJM](http://www.nejm.org/)
- [Genome Research](http://genome.cshlp.org/)
- [Nature biotechnology](http://www.nature.com/nbt/index.html)
- [Nature Review Genetics](http://www.nature.com/nrg/index.html)
- [Nature Review Cancer](http://www.nature.com/nrc/index.html)
- [PLOS Computational Biology](http://www.ploscompbiol.org/)
- [Annual Review](http://www.annualreviews.org/)



### Courses

- [龙星计划2013-翁志萍](http://www.youku.com/playlist_show/id_19532648.html)
- [Statistical methods and data mining in microarray analysis-George C. Tseng](http://www.biostat.pitt.edu/biost2055/11/)
- [Analyzing High Throughput Sequencing Data-Istvan Albert](http://www.personal.psu.edu/iua1/courses/courselist.html)
- [Applied Bioinformatics-Istvan Albert](http://www.personal.psu.edu/iua1/courses/2014-BMMB-852.html)
- [NGS Data Analysis Course](http://ngscourse.github.io/)
- [SSC Intro to NGS Bioinformatics Course-University of Texas](https://wikis.utexas.edu/display/bioiteam/SSC+Intro+to+NGS+Bioinformatics+Course)
- [Yunlong Liu](http://compbio.iupui.edu/group/1/pages/next_gen_course)
- [Computational Biology-Shirley Xiaole Liu](http://www.stat115.org/index.html)
- [Hot topics-MIT](http://jura.wi.mit.edu/bio/education/hot_topics/)
- [Biological Statistic](http://udel.edu/~mcdonald/statintro.html)
- [ROSALIND](http://rosalind.info/problems/locations/)
- [简明python教程](http://woodpecker.org.cn/abyteofpython_cn/chinese/index.html)
- [Machine Learning-Andrew Ng](http://cs229.stanford.edu/)
- [Machine Learning-Bo(Eric) Xing](http://www.cs.cmu.edu/~epxing/Class/10701/)
- [Workshop of Bioinformatic.ca](http://bioinformatics.ca/workshops)
- [edx](https://www.edx.org/)
- [coursera](https://www.coursera.org/)
	- [Data Structures and Algorithms Specializations by UCSD](https://www.coursera.org/specializations/data-structures-algorithms)
		- Algorithmic Toolbox
		- Data Structures
		- Algorithms on Graphs
		- Algorithms on Strings
		- Advanced Algorithms and Complexity
		- Genome Assembly Programming Chanllenge
	- [Bioinformatics Specializations by UCSD](https://www.coursera.org/specializations/bioinformatics)
		- Finding Hidden Messages in DNA
		- Genome Sequencing
		- Comparing Genes, Proteins, and Genomes
		- Deciphering Molecular Evolution
		- Genomic Data Science and Clustering
		- Finding Mutations in DNA and Proteins
		- Bioinformatics Capstone: Big Data in Biology
	- Bioinformatics Methods [I](https://www.coursera.org/learn/bioinformatics-methods-1) and [II](https://www.coursera.org/learn/bioinformatics-methods-2)
- [Genetic Epidemiology](http://www.montefiore.ulg.ac.be/~kvansteen/TeachingGEUA20142015final.html)
- [An Online Bioinformatics Curriculum by Plos Computational Biology](http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1002632)
- [A New Online Computational Biology Curriculum](http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1003662)

### Project

- [HapMap](http://hapmap.ncbi.nlm.nih.gov/)
- [ENCODE](http://genome.ucsc.edu/ENCODE/)
- [TCGA](http://cancergenome.nih.gov/)
- [RoadMap Epigenomics Project](http://www.roadmapepigenomics.org/)
- [BLUEPRINT](http://www.blueprint-epigenome.eu/)
- [IHEC](http://ihec-epigenomes.org/)
- [1000 Genomes](http://www.1000genomes.org/)
- [TARGET](http://target.nci.nih.gov/dataMatrix/)
- [GTEx](http://www.gtexportal.org/home/): Genotype-Tissue Expression
- [Fantom5](http://fantom.gsc.riken.jp/5/)
- [CCLE](http://www.broadinstitute.org/ccle/home)
- [HMP](http://hmpdacc.org/)
- [The Wellcome Trust UK10K project](http://www.uk10k.org/)
- [Exac](http://exac.broadinstitute.org/): Exome Aggregation Consortium

![](/images/timeline_project.jpg)

### Funding

- [NIH](http://www.nih.gov/)
- [NCI](http://www.cancer.gov/)
- [NHGRI](https://www.genome.gov/)
- [国自然](http://www.nsfc.gov.cn/)
- [科技部](http://www.most.gov.cn/kjjh/xmsb/)

### Database

- [NCBI](http://www.ncbi.nlm.nih.gov/)
- [GEO](http://www.ncbi.nlm.nih.gov/geo/)
- [Array-express](http://www.ebi.ac.uk/arrayexpress/)
- [SRA](http://www.ncbi.nlm.nih.gov/sra/)
- [GenBank]( http://www.ncbi.nlm.nih.gov/genbank/)
- [PDB]( http://www.rcsb.org/pdb/home/home.do)
- [GO (Gene Ontology) ](http://geneontology.org/)
- [SO (Sequence Ontology)](http://www.sequenceontology.org/)
- [KEGG](http://www.genome.jp/kegg/)
- [UCSC Genome Browser](http://genome.ucsc.edu/)
- [Ensembl](http://www.ensembl.org/index.html)
- [dbSNP](http://www.ncbi.nlm.nih.gov/snp/)
- [COSMIC](http://www.sanger.ac.uk/genetics/CGP/cosmic/)
- [ClinVar](http://www.ncbi.nlm.nih.gov/clinvar/): genomic variation and its relationship to human health
- [OMIM](http://omim.org/): Online Mendelian Inheritane in Man
- [primer design database](http://www.hsls.pitt.edu/obrc/index.php?page=pcr_oligos)
- [ExAC](http://exac.broadinstitute.org/)
- [Omictools](http://omictools.com/)
- [HGMD](http://www.hgmd.cf.ac.uk/ac/index.php): Human Gene Mutation Database

### Bioinformatic cloud

- [Seven Bridge](https://www.sbgenomics.com/)
- [DNAnexus](https://www.dnanexus.com/)
- [Google Genomics](https://cloud.google.com/genomics/)
- [NCI's Genomic Data Commons](https://gdc.nci.nih.gov/)
- [Global Alliance for Genomics and Health](http://ga4gh.org)

### Your favorite blog on Biostar

- Your Favorite Bioinformatics Blogs (March 2010)
- Your Favorite Bioinformatics Blogs (October 2011)
- What Are Your Favorite Bioit News Resources / Blogs (2012 Edition)
- Your Favorite Bioinformatics Blog(S) (2013 Edition)

### 微信公众账号
- 赛先生
- 知识分子
- 测序中国
- 丁香园
- 生物谷
- 知因knowgene
- 博士圈Illumina


## Computer science

### Blogs

- [阮一峰]( http://www.ruanyifeng.com/blog/)
- [Chengwei's Words](http://www.chengweiyang.cn/archives/)
- [Mozillazg's Blog](https://mozillazg.com/)
- [James Hamilton - VP at Amazon Web Services](http://perspectives.mvdirona.com/)
- [Denny Britz, a engineer in Google](http://blog.dennybritz.com/)
- [Linux工具快速教程](http://linuxtools-rst.readthedocs.io/zh_CN/latest/index.html) 

### Courses

- [《Introduction to Computer Science and Programming Using Python》](https://www.edx.org/course/introduction-computer-science-mitx-6-00-1x-9) by MIT on edx
- [《Introduction to Computational Thinking and Data Science》](https://www.edx.org/course/introduction-computational-thinking-data-mitx-6-00-2x-5) by MIT on edx
- 《Algorithms》[part I](https://www.coursera.org/learn/algorithms-part1#) and [part II](https://www.coursera.org/learn/algorithms-part2#) by Princeton on coursera
- [《Algorithms Specialization》](https://www.coursera.org/specializations/algorithms) by Stanford on coursera
- [《Build Complete Web Solutions》](https://www.coursera.org/specializations/full-stack) by HST on coursera
- [《Learn to Program and Analyze Data with Python》](https://www.coursera.org/specializations/python) by University of Michigan on coursera
- [《Become a Rails Developer》](https://www.coursera.org/specializations/ruby-on-rails) by JHU on couresra
- [《Introduction to Computer Systems》](http://www.cs.cmu.edu/afs/cs/academic/class/15213-f14/www/index.html) by CMU
- [《CMU CS Bachelors Curriculum》](https://www.csd.cs.cmu.edu/content/bachelors-curriculum-admitted-2014)
- [《Parallel Computer Architecture and Programming》](http://15418.courses.cs.cmu.edu/spring2016/home) by CMU
- [《Computer Networks》](http://www.cs.cmu.edu/~prs/15-441-F14/index.html) by CMU
- [《CS155 Computer and Network Security》](https://crypto.stanford.edu/cs155/syllabus.html) by Stanford

### Books

- [Pro Git](https://book.douban.com/subject/3420144/)
- [HTTP: The Definitive Guide](https://book.douban.com/subject/1440226/)
- [HTML and CSS: Design and Build Websites](https://book.douban.com/subject/6585090/)
- [JavaScript and JQuery: Interactive Front-end Web Development](https://book.douban.com/subject/20022383/)
- [Beginning Linux Programming](https://book.douban.com/subject/2361237/)
- [Python Requests Essentials](https://book.douban.com/subject/26428434/)
- [Thinking in Java](https://book.douban.com/subject/1474824/)
- [图解HTTP](https://book.douban.com/subject/25863515/)
- [Learning Python](https://book.douban.com/subject/22139956/)
- [C Primer Plus](https://book.douban.com/subject/25802313/)
- [The C Programming Language](https://book.douban.com/subject/1236999/)
- [Mastering Sublime Text](https://book.douban.com/subject/25804133/)
- [Practical Vim](https://book.douban.com/subject/10599776/)
- [Effective Python](https://book.douban.com/subject/26312313/)

### Tools

- [Jupyter](http://jupyter.org/)


